code/modules/mob/living/silicon/laws.dm

```diff
@@ -14,9 +14,12 @@
 	throw_alert("newlaw", /atom/movable/screen/alert/newlaw)
 	if(announce && last_lawchange_announce != world.time)
 		to_chat(src, "<b>Your laws have been changed.</b>")
+		overlay_fullscreen("law_change", /atom/movable/screen/fullscreen/law_change, 1)
 		// lawset modules cause this function to be executed multiple times in a tick, so we wait for the next tick in order to be able to see the entire lawset
 		addtimer(CALLBACK(src, PROC_REF(show_laws)), 0)
 		addtimer(CALLBACK(src, PROC_REF(deadchat_lawchange)), 0)
+		// Wait a tick and clear the vignette
+		addtimer(clear_fullscreen("law_change"), 10)
```

Comment:
This is now how timers work, there is a difference between providing the return value of a function call and a function reference.
When you write clear_fullscreen("law_change"), you immedaitely invoke the clear_fullscreen function and pass the return value of that function into the first parameter of addtimer. What you need is a function reference, which doesn't call the function but instead allows it to be called. This is what CALLBACK is for.
CALLBACK(src, TYPED_PROC_REF(/client/or_whatever, clear_fullscreen), "law_change")

Reply:
*not how

Reply:
Whoops

Reply:
Roger. Will resolve this when I get home from work tonight.
Thanks for putting an example there, still fairly new to dm.

Reply:
> TYPED_PROC_REF(/client/or_whatever, clear_fullscreen), "law_change")
Since clear_fullscreen is a global method, I'm going to try 
addtimer(CALLBACK(src, PROC_REF(clear_fullscreen), "law_change"), 5)

Reply:
Updated the code per request. I noticed that the behaviour changed, and it now waits before clearing, depending on the number of deciseconds set.

Reply:
Yea, that is because before you were instantly calling the function by using clearfull_screen(), so there isn't much point of using a timer. Now you are passing a callback to the timer, which is essentially like a delayed function. It tells the program about a function it can execute, but doesn't execute it (the timer now knows how to execute clear_fullscreen so that when it finishes, it executes that function). Since you are using addtimer properly now, thats why there is a short delay before clearing.